src/Test/Test.csproj

```diff
@@ -163,4 +163,7 @@
   <Import Project="..\packages\MSTest.TestAdapter.1.1.18\build\net45\MSTest.TestAdapter.targets" Condition="Exists('..\packages\MSTest.TestAdapter.1.1.18\build\net45\MSTest.TestAdapter.targets')" />
   <Import Project="..\packages\Accord.3.8.0\build\Accord.targets" Condition="Exists('..\packages\Accord.3.8.0\build\Accord.targets')" />
   <Import Project="..\packages\NBench.1.2.2\build\NBench.targets" Condition="Exists('..\packages\NBench.1.2.2\build\NBench.targets')" />
+  <PropertyGroup>
+    <PreBuildEvent>xcopy /y "$(SolutionDir)Library\bin\$(Configuration)\net45\*.dll" "$(TargetDir)"</PreBuildEvent>
```

Comment:
My biggest concern is this one - it has a ProjectReference to the Library project, so it should already be present. Do you have repro steps for the failure? It might be an issue in the test runner for instance.

Reply:
I can reproduce by doing the following:

- Make a clone into a clean directory
- Build all projects (Build->Build solution in VS)

I then get the following output from the build:
![image](https://user-images.githubusercontent.com/9482927/74970533-d0612180-541e-11ea-817b-8f627283c099.png)

```
1>------ Build started: Project: Library, Configuration: Debug Any CPU ------
1>Library -> D:\Users\erik\Source\Repos\FastLinq\src\Library\bin\Debug\net45\FastLinq.dll
1>Library -> D:\Users\erik\Source\Repos\FastLinq\src\Library\bin\Debug\netcoreapp20\FastLinq.dll
1>Library -> D:\Users\erik\Source\Repos\FastLinq\src\Library\bin\Debug\netcoreapp11\FastLinq.dll
1>Successfully created package 'D:\Users\erik\Source\Repos\FastLinq\src\Library\bin\Debug\FastLinq.0.4.0.nupkg'.
2>------ Build started: Project: Test, Configuration: Debug Any CPU ------
3>------ Build started: Project: Benchmark, Configuration: Debug Any CPU ------
2>CSC : error CS2001: Source file 'D:\Users\erik\Source\Repos\FastLinq\src\Test\IntArgument_SameResultAsEnumerable_Tests.cs' could not be found.
2>CSC : error CS2001: Source file 'D:\Users\erik\Source\Repos\FastLinq\src\Test\Boolean_Func_SameResultAsEnumerable_Tests.cs' could not be found.
2>CSC : error CS2001: Source file 'D:\Users\erik\Source\Repos\FastLinq\src\Test\SingleGenericParameter_SingleInput_SameResultAsEnumerable_Tests.cs' could not be found.
3>Benchmarks\CastBenchmark.cs(315,21,315,66): warning CS0618: 'FastLinq.Cast<TResult>(ICollection)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(327,21,327,54): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(339,21,339,53): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(351,21,351,54): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\RepeatBenchmark.cs(82,25,82,29): warning CS0219: The variable 'item' is assigned but its value is never used
3>Benchmarks\CastList2Benchmark.cs(58,29,59,33): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\EnumerableWithCountBenchmark.cs(101,27,101,31): warning CS0169: The field 'List.list' is never used
3>Benchmarks\EnumerableWithCountBenchmark.cs(106,30,106,40): warning CS0169: The field 'HashSet.collection' is never used
3>Benchmarks\RepeatBenchmark.cs(82,25,82,29): warning CS0219: The variable 'item' is assigned but its value is never used
3>Benchmarks\CastBenchmark.cs(315,21,315,66): warning CS0618: 'FastLinq.Cast<TResult>(ICollection)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(327,21,327,54): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastList2Benchmark.cs(58,29,59,33): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(339,21,339,53): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(351,21,351,54): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\EnumerableWithCountBenchmark.cs(101,27,101,31): warning CS0169: The field 'List.list' is never used
3>Benchmarks\EnumerableWithCountBenchmark.cs(106,30,106,40): warning CS0169: The field 'HashSet.collection' is never used
3>Benchmark -> D:\Users\erik\Source\Repos\FastLinq\src\Benchmark\bin\Debug\netcoreapp2.0\Benchmark.dll
3>Benchmark -> D:\Users\erik\Source\Repos\FastLinq\src\Benchmark\bin\Debug\netcoreapp1.1\Benchmark.dll
3>Done building project "Benchmark.csproj".
3>Benchmarks\CastList2Benchmark.cs(58,29,59,33): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(315,21,315,66): warning CS0618: 'FastLinq.Cast<TResult>(ICollection)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(327,21,327,54): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(339,21,339,53): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\CastBenchmark.cs(351,21,351,54): warning CS0618: 'FastLinq.Cast<T>(IList)' is obsolete: 'Prefer Cast<T, TOther> whenever possible.'
3>Benchmarks\RepeatBenchmark.cs(82,25,82,29): warning CS0219: The variable 'item' is assigned but its value is never used
3>Benchmarks\EnumerableWithCountBenchmark.cs(101,27,101,31): warning CS0169: The field 'List.list' is never used
3>Benchmarks\EnumerableWithCountBenchmark.cs(106,30,106,40): warning CS0169: The field 'HashSet.collection' is never used
3>Benchmark -> D:\Users\erik\Source\Repos\FastLinq\src\Benchmark\bin\Debug\net46\Benchmark.exe
3>Done building project "Benchmark.csproj".
========== Build: 2 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

Looking in Process Monitor I can see that first FastLinq.dll is built (VBCSCompiler.exe) and then Visual Studio (devenv.exe) tries to access `FastLinq\src\Library\bin\Debug\netcoreapp11\FastLinq.dll`and fails.
![image](https://user-images.githubusercontent.com/9482927/74971037-b5db7800-541f-11ea-9375-661fee5c79e0.png)

If I right-click and select Debug T4 template on `Boolean_Func_SameResultAsEnumerable_Tests.tt` I get the following error message before debugging starts:
![image](https://user-images.githubusercontent.com/9482927/74971187-f2a76f00-541f-11ea-94fa-4c6bc9a3496c.png)

Reply:
Hence it seems like each T4 template can be considered its own project with its own references and it will be built before the rest of the project using only the references at the top of the files.
```
<#@ assembly name="$(TargetDir)FastLinq.dll" alias="FL" #>
```

However I still got this warning after adding the file copy as a pre-build task. Will investigate more:
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning		A custom tool 'TextTemplatingFileGenerator' is associated with file 'IntArgument_SameResultAsEnumerable_Tests.tt', but the output of the custom tool was not found in the project.  You may try re-running the custom tool by right-clicking on the file in the Solution Explorer and choosing Run Custom Tool.	Test
```

Reply:
I will continue debugging tomorrow evening, [it seems](https://stackoverflow.com/questions/1646580/get-visual-studio-to-run-a-t4-template-on-every-build) like T4 templates aren't run at compile time as a default.

Reply:
Thanks, the t4 templates were me using the project as a playground :) I'll take a look at the steps you used and figure out what's wrong in my setup.

Reply:
Okay, got it all working now, I don't see errors for the Framework versions, but I know yours look more correct anyway :) Could you revert this one change, I'll send a PR for you to confirm my fix, and I'll accept your change for the Frameworks & Microsoft.CSharp warnings?

Reply:
Removed the pre-build task. I had never used T4 before so I saw this as a chance to try it out as well :-)